### ISSUE TITLE
Add support for concatMapEager

### ISSUE THREAD

**Issue by rochdev** [ReactiveX/rxjs#2733]
When calling concatMap, the handler is called in series so it has to return a value before entering the next iteration.

In some cases it would be more efficient to be able to run the handler in parallel and then drain the results in order, especially if the handler is waiting for IO.

See http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#concatMapEager(io.reactivex.functions.Function)

**Comment by kwonoj** [comment]
it is need to be merged but doc itself is ready - please refer https://github.com/ReactiveX/rxjs/pull/2524 .

in general we recommend to create new operator as user level code / packages except very few cases to avoid api surface issues - for this one too I think this operator can be implemented without internals of rx codebase itself.

**Comment by rochdev** [comment]
I would argue that I expected the behaviour of `concatMap` to be what is done by `concatMapEager`. Would it be possible maybe to simply add a concurrency option to `concatMap` so a new operator is not introduced? (with a default of 1 to preserve backward compatibility)

**Comment by kwonoj** [comment]
For your argument, some other people might say they expect completely different behavior. 

Clarify, point I'd like to share in this issue is not about the usefulness of operator suggested. It is more like due to it is impossible for us to create core API set satisfies all operator requirements, we're trying to make core API surface minimal as possible, including ongoing discussions for breaking changes of removing operator from core set, and recommend to opt-in extended, custom operator features. As suggested above, any custom operator can be published as user level as composable manner, which functionally should not differ. Or either if you believe this should be included in core, we would like to have a consensus about reasoning.

**Comment by rochdev** [comment]
While I agree with your general mindset, this is already implemented in the Java version. It doesn't make sense to me to have a different core API for each implementation. This feature would basically be part of some kind of `rxjava-for-rxjs` extension. So while I don't necessarily believe it should be included in core, it is already included in core in other implementations so I would expect a consistent experience across languages.

**Comment by kwonoj** [comment]
Each rxjs binding for language already have different set of operator and it is not a goal to provide same set of operator between languages. You can compare different rx language binding to see those differences.

**Comment by kwonoj** [comment]
I'm closing for this now.
